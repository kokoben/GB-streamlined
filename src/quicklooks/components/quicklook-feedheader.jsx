import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import FeedHeader from '../../components/feed-header';
import { fetchQuicklookSearchVideos, setQuicklookVideo } from '../actions/index';
import QlSearchResultsSelector from '../selectors/current-ql-search-results';

const QuicklookFeedHeader = props => (
  <FeedHeader
    category="Quicklook"
    placeholder="Search for Quick Look videos..."
    fetchVideos={props.fetchQuicklookSearchVideos}
    currentResults={props.currentQlSearchResults}
    setVideo={props.setQuicklookVideo}
  />
);

/* eslint-disable react/forbid-prop-types */
QuicklookFeedHeader.propTypes = {
  fetchQuicklookSearchVideos: PropTypes.func.isRequired,
  currentQlSearchResults: PropTypes.array.isRequired,
  setQuicklookVideo: PropTypes.func.isRequired,
};
/* eslint-enable */

const mapStateToProps = state => ({
  currentQlSearchResults: QlSearchResultsSelector(state),
});

const mapDispatchToProps = dispatch => (
  bindActionCreators({
    fetchQuicklookSearchVideos,
    setQuicklookVideo,
  }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(QuicklookFeedHeader);
