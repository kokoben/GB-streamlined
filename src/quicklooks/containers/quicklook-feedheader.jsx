import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import FeedHeader from '../../components/feed-header';
import { fetchQuicklookSearchVideos } from '../actions/index';
import QlSearchResultsSelector from '../selectors/current-ql-search-results';

class QuicklookFeedHeader extends Component {
  render() {
    return (
      <FeedHeader
        category="Quicklook"
        placeholder="Search for Quick Look videos..."
        fetchVideos={this.props.fetchQuicklookSearchVideos}
        currentResults={this.props.currentQlSearchResults}
      />
    );
  }
}

/* eslint-disable react/forbid-prop-types */
QuicklookFeedHeader.propTypes = {
  fetchQuicklookSearchVideos: PropTypes.func.isRequired,
  currentQlSearchResults: PropTypes.array.isRequired,
};
/* eslint-enable */

const mapStateToProps = state => ({
  currentQlSearchResults: QlSearchResultsSelector(state),
});

const mapDispatchToProps = dispatch => (
  bindActionCreators({
    fetchQuicklookSearchVideos,
  }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(QuicklookFeedHeader);
