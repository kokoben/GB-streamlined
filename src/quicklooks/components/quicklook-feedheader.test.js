import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { QuicklookFeedHeader } from './quicklook-feedheader';

describe('quicklook-feedheader tests', () => {
  it('renders correctly', () => {
    const mockFetchQuicklookSearchVideos = jest.fn();
    const mockSetQuicklookVideo = jest.fn();
    const mockSetVideo = jest.fn();

    const wrapper = shallow(<QuicklookFeedHeader
      quicklookSearchResults={[]}
      currentQlSearchResults={[]}
      fetchQuicklookSearchVideos={mockFetchQuicklookSearchVideos}
      setQuicklookVideo={mockSetQuicklookVideo}
      setVideo={mockSetVideo}
    />);

    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});

