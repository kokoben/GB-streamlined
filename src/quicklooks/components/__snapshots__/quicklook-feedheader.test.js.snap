// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quicklook-feedheader tests renders correctly 1`] = `
<FeedHeader
  category="Quicklook"
  currentResults={Array []}
  fetchVideos={[Function]}
  placeholder="Search for Quick Look videos..."
  results={Array []}
  setVideo={[Function]}
/>
`;
